web/app/js/components/ConfigureProfilesMsg.jsx

```diff
@@ -89,48 +117,83 @@ class ConfigureProfilesMsg extends React.Component {
           variant="outlined"
           color="primary"
           size="small"
-          onClick={this.handleClickOpen}>Create Service Profile
+          onClick={this.handleClickOpen}>
+          Create Service Profile
         </Button>
       );
     }
+
+
+    let disableDownloadButton = _isEmpty(query.service) || _isEmpty(query.namespace) ||
+      error.service || error.namespace;
+    let downloadButton = (
+      <Button
+        disabled={disableDownloadButton}
+        onClick={() => this.handleClose(downloadUrl)}
+        color="primary">
+        Download
+      </Button>
+    );
+
     return (
       <React.Fragment>
         {button}
-
         <Dialog
           open={this.state.open}
           onClose={this.handleClose}
           aria-labelledby="form-dialog-title">
           <DialogTitle id="form-dialog-title">New service profile</DialogTitle>
           <DialogContent>
             <DialogContentText>
-              To create a service profile, download a profile and then apply it with `kubectl apply`.
+              To create a service profile, download a profile and then apply it
+              with `kubectl apply`.
             </DialogContentText>
-            <TextField
-              id="service"
-              label="Service"
+            <FormControl
               className={classes.textField}
-              value={this.state.service}
-              onChange={this.handleChange('service')}
-              margin="normal" />
-            <TextField
-              id="namespace"
-              label="Namespace"
+              onBlur={() => this.validateFields('service', query.service)}
```

Comment:
using `onChange` here instead of `onBlur` will provide more immediate feedback to the user